jellyfin_kodi/objects/kodi/kodi.py

```diff
@@ -123,46 +132,59 @@ def add_thumbnail(person_id, person, person_type):
 
                 self.artwork.update(person['imageurl'], person_id, art, "thumb")
 
-        def add_link(link, person_id):
-            self.cursor.execute(QU.update_link.replace("{LinkType}", link), (person_id,) + args)
-
         cast_order = 1
 
+        bulk_updates = {}
+
         for person in people:
             person_id = self.get_person(person['Name'])
 
             if person['Type'] == 'Actor':
-
+                sql = QU.update_actor
                 role = person.get('Role')
-                self.cursor.execute(QU.update_actor, (person_id,) + args + (role, cast_order,))
+                bulk_updates.setdefault(sql, []).append((person_id,) + args + (role, cast_order,))
                 cast_order += 1
 
             elif person['Type'] == 'Director':
-                add_link('director_link', person_id)
+                sql = QU.update_link.replace("{LinkType}", 'director_link')
+                bulk_updates.setdefault(sql, []).append((person_id,) + args)
 
             elif person['Type'] == 'Writer':
-                add_link('writer_link', person_id)
+                sql = QU.update_link.replace("{LinkType}", 'writer_link')
+                bulk_updates.setdefault(sql, []).append((person_id,) + args)
 
             elif person['Type'] == 'Artist':
-                add_link('actor_link', person_id)
+                sql = QU.update_link.replace("{LinkType}", 'actor_link')
+                bulk_updates.setdefault(sql, []).append((person_id,) + args)
 
             add_thumbnail(person_id, person, person['Type'])
 
+        for sql, parameters in bulk_updates.items():
+            self.cursor.executemany(sql, parameters)
+
     def add_person(self, *args):
 
         person_id = self.create_entry_person()
         self.cursor.execute(QU.add_person, (person_id,) + args)
 
         return person_id
 
-    def get_person(self, *args):
-
+    def _get_person(self, *args):
         try:
+            return self.add_person(*args)
+        except IntegrityError:
+            # The person already exists in the database
+            # Now we can do the expensive operation of fetching the id
```

Comment:
Shouldn't this block also be checking `_people_cache` first, and then doing the add if they don't already exist?  Similar to the `get_person` below

Reply:
_get_person is internal/private. Ideally, it's an implementation detail which ideally should only be called by get_person.
If another callee calls this directly then I can only assume that they do not want to access the cache. This is the only case where the catch block should get called as calling _get_person via get_person means that the person does not exist in the db and add_person should'nt raise an IntegrityError

Reply:
I see what you mean now.  I should probably look at the full file instead of just the little individual changed snippets